back/engines/commercial/analysis/app/models/analysis/analysis.rb

```diff
@@ -87,9 +87,10 @@ def project_xor_phase_present
     # The linked project or phase should be a valid form context (the
     # participation_context the custom_form is associated with)
     def project_or_phase_form_context
```

Comment:
@kogre This part is still todo. It was indirectly doing calls to `ideation?`, `voting?` etc. which we should ideally avoid. I just temporarily rewrote a more explicit version here, and would like to come up with good predicates in the participation method through which this can be expressed. Maybe `supports_analysis_on_phase?` and `supports_analysis_on_project`?

Reply:
We decided to go with simple solutions for this PR, and added a separate task for ensuring that the analysis feature works well for proposals.